### PR TITLE
Fix some gchar allocation leaks

### PR DESCRIPTION
**Pull request by kmilos**
Found using cppcheck 2.9

### CODE REVIEW

File: src/gui/styles_dialog.c [hunk 1]
```diff
@@ -705,20 +705,24 @@ GtkWidget *dt_gui_style_content_dialog(char *name, const int imgid)
   GtkWidget *label = NULL;
 
   // name
-  snprintf(buf, sizeof(buf), "<b>%s</b>", g_markup_escape_text(name, -1));
+  gchar *esc_name = g_markup_escape_text(name, -1);
```
Comment (dterrahe) [inline src/gui/styles_dialog.c:708]: Could to this in one go with `g_markup_printf_escaped` (without need for buf)

File: src/iop/channelmixerrgb.c [hunk 2]
```diff
@@ -3743,11 +3743,11 @@ void _auto_set_illuminant(dt_iop_module_t *self, dt_dev_pixelpipe_t *pipe)
   dt_Lab_2_LCH(Lab, Lch);
 
   // Write report in GUI
+  gchar *str = g_strdup_printf(_("L: \t%.1f %%\nh: \t%.1f °\nc: \t%.1f"), Lch[0], Lch[2] * 360.f, Lch[1]);
   ++darktable.gui->reset;
```
Comment (dterrahe) [inline src/iop/channelmixerrgb.c:3747]: No need to bracket pure gtk calls in `++/--darktable.gui->reset;` 
Can be cleaned up.

File: src/iop/exposure.c [hunk 3]
```diff
@@ -725,10 +725,11 @@ static void _auto_set_exposure(dt_iop_module_t *self, dt_dev_pixelpipe_t *pipe)
   dt_Lab_2_LCH(Lab, Lch);
 
   // Write report in GUI
+  gchar *str = g_strdup_printf(_("L : \t%.1f %%"), Lch[0]);
   ++darktable.gui->reset;
```
Comment (dterrahe) [inline src/iop/exposure.c:729]: Same

File: src/gui/styles_dialog.c (hunk 1 above)
Comment (kmilos) [reply src/gui/styles_dialog.c:708]: Thanks for the pointer, will do.

File: src/iop/channelmixerrgb.c (hunk 2 above)
Comment (kmilos) [reply src/iop/channelmixerrgb.c:3747]: It's in a quite a few places actually. As it is not related to leaks, I'd prefer that cleanup happens in its own, separate PR.

Comment (dterrahe) [reply src/iop/channelmixerrgb.c:3747]: I should have been more specific; pure gtk calls _without side effects_. So for example `gtk_toggle_button_set_active` will usually trigger a signal which may need to be deactivated via darktable.gui->reset.

I believe this particular instance was a leftover from a refactoring where previously the gui->reset bracketed a bunch of calls-with-sideeffects but that is no longer the case. No need to clean up now, if you want to keep the PR to its essentials.

**Review by TurboGit — APPROVED** [review]
Looks good, thanks!

File: src/gui/styles_dialog.c (hunk 1 above)
Comment (kmilos) [reply src/gui/styles_dialog.c]: Done

**Comment by dterrahe** [comment]
> Found using cppcheck 2.9

Actually surprised it didn't find more. Has this scan been performed in the past?

**Comment by kmilos** [comment]
> Actually surprised it didn't find more. Has this scan been performed in the past?

Yes, I found a couple of more similar leaks in the 4.0 release cycle: https://github.com/darktable-org/darktable/pull/11044

There quite a few more other "errors", but could be false positives, or not something I feel confident analyzing and fixing, so I went for the low hanging fruit first...

**Comment by dterrahe** [comment]
> There are quite a few more other "errors"

Does it generate a log file that you could post here for others to peruse? It seems to take a while to run...

**Comment by kmilos** [comment]
Takes a while indeed... Here are the results from my run:

[darktable-cppcheck.zip](https://github.com/darktable-org/darktable/files/10447817/darktable-cppcheck.zip)

**Comment by dterrahe** [comment]
Thanks!

It doesn't like libraw and rawspeed much.
Looks like some code quality improvements could reduce the number of false positives.
And some warnings might be an indication of real problems (but I wouldn't necessarily be able to tell/test).
It would be useful for others to have a look. I'm thinking of @jenshannoschwalm @dtorop (and @TurboGit obviously) for example.

**Comment by kmilos** [comment]
Rawspeed reports could be due to my settings (e.g. C++11 rather than 17), and I should've excluded src/external anyway...

**Comment by TurboGit** [comment]
The report is intimidating :)

At least there is some code we can remove when you look at the `identicalInnerCondition` or `nullPointerRedundantCheck`.

I've also spotted `unusedAllocatedMemory` - indeed a chunk of memory allocated but never used.

Now what to do?

I can handle the 3 error id above (seems like easy to do).

We can then dispatch other error id to different devs. @kmilos : If you could handle this it would be nice... You could create issues there and assigned to different devs, you can count me on this. Would that work for you?

**Comment by kmilos** [comment]
Sorry, but I'm going to have to sit this one out 😕

**Comment by TurboGit** [comment]
> Sorry, but I'm going to have to sit this one out confused

No problem. I'll merge this and will probably fix some issue reported in the cppcheck report. But looking at it more closely there is many false positive... So probably too much work for a little gain.

**Review by TurboGit — APPROVED** [review]
Thanks.

**Comment by kmilos** [comment]
@TurboGit I might still fix a few here or there, but unfortunately can't take on responsibility of "project managing" this...

**Comment by TurboGit** [comment]
> @TurboGit I might still fix a few here or there, but unfortunately can't take on responsibility of "project managing" this...

No problem. We all have limited time and I can understand. I'll also may find some time to look at some others issues.

**Comment by jenshannoschwalm** [comment]
@TurboGit i you like you can give me a section of the log to work on ...

**Pull request merged**